### PR TITLE
WIP spires: support for particles

### LINKED ISSUE #38: search: support for particles
We should have a whitelist of particles and formulas that are pre-extracted and should not interfere with the regular parsing.

See details into inveniosoftware/invenio/699

### PR DESCRIPTION
**Pull request by Panos512**
- Adds a list of particles.
- Identifies particles as "particle" instead of "value".
  (closes #38)

Signed-off-by: Panos Paparrigopoulos panos.paparrigopoulos@cern.ch

### CODE REVIEW

File: invenio_query_parser/config.py [hunk 1]
```diff
@@ -0,0 +1,783 @@
+# -*- coding: utf-8 -*-
```
Comment (jirikuncar) [inline invenio_query_parser/config.py]: Isn't `spires` contrib config enough?

File: invenio_query_parser/contrib/spires/config.py [hunk 2]
```diff
@@ -209,3 +209,763 @@
     'refs': 'refersto',
     'refersto': 'refersto',
 }
+
+
+PARTICLES = {
```
Comment (jirikuncar) [inline invenio_query_parser/contrib/spires/config.py]: `pip install autopep8 && autopep8 -i invenio_query_parser/contrib/spires/config.py`

File: invenio_query_parser/ast.py [hunk 3]
```diff
@@ -163,6 +163,8 @@ class Keyword(Leaf):
 class Value(Leaf):
     pass
 
+class Particle(Leaf):
```
Comment (jirikuncar) [inline invenio_query_parser/ast.py]: this is `spires` specific `Leaf`

File: invenio_query_parser/contrib/spires/walkers/pypeg_to_ast.py [hunk 4]
```diff
@@ -64,6 +64,10 @@ def visit(self, node, child):
     def visit(self, node):
         return ast.Value(node.value)
 
+    @visitor(parser.Particle)
+    def visit(self, node):
+        return ast.Particle(node.value)
```
Comment (jirikuncar) [inline invenio_query_parser/contrib/spires/walkers/pypeg_to_ast.py]: Shouldn't we return something like:
```python
ast.KeywordValue(ast.Keyword(config.PARTICLE_KEYWORD), ast.Value(node.value))
```

**Comment by jirikuncar** [comment]
@Panos512 all changes should touch only `invenio_query_parser/contrib/spires/` folder.

**Comment by Panos512** [comment]
@jirikuncar Thanks for the suggestions! I will get back to it as soon as possible.

**Comment by jmartinm** [comment]
ping @Panos512 what was the status of this PR?

**Comment by Panos512** [comment]
@jmartinm I can pick it up again if we still need it.

**Comment by egabancho** [comment]
@kaplun you mentioned Panos left already and this PR has been sleeping for a year or so, do you think you'll have time to take look at it should we close it for now?

**Comment by jacquerie** [comment]
No, I'm going to take care of it.

**Comment by lnielsen** [comment]
Closing PR as it's been stale for a long time.

**Pull request closed without merging**